### ISSUE TITLE
Make  vcard version of auto backup of contacts compatible with Nextcloud contacts (min vCard 3.0)

### ISSUE THREAD

**Issue by Bugsbane** [nextcloud/android#4727]
### Actual behaviour
Contacts exported by the automatic backup are all reported as being vCard version 2.1 which is incompatible with cardDav servers, and thus show as needing fixing by the Nextcloud contacts app. (See the start of the error message in the log)

### Expected behaviour
VCF file exported by automatic backup should import without error into the Nextcloud contacts app.
 
### Steps to reproduce
1. Turn on automatic contact backups in the Nextcloud android app
2. Choose import from the Nextcloud contacts app

### Environment data
Android version:
9

Device model: 
Samsung Galaxy A8

Stock or customized system:
Stock

Nextcloud app version:
3.7.0

Nextcloud server version:
15.0.10

#### Nextcloud error log
``` 

[webdav] Fatal: Sabre\DAV\Exception\UnsupportedMediaType: Validation error in vCard: CardDAV servers are not allowed to accept vCard 2.1. at <<closure>>

 0. /var/www/nextcloud/3rdparty/sabre/dav/lib/CardDAV/Plugin.php line 316
    validateVCard("BEGIN:VCARD\r\n ... n", false)
 1. <<closure>>
    beforeCreateFile("addressbooks/us ... f", "BEGIN:VCARD\r\n ... n", OCA\DAV\CardDAV\AddressBook {}, false)
 2. /var/www/nextcloud/3rdparty/sabre/event/lib/EventEmitterTrait.php line 105
    call_user_func_array([OCA\DAV\CardDAV ... "], ["addressbooks/u ... e])
 3. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 1094
    emit("beforeCreateFile", ["addressbooks/u ... e])
 4. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php line 525
    createFile("addressbooks/us ... f", "BEGIN:VCARD\r\n ... n", null)
 5. <<closure>>
    httpPut(Sabre\HTTP\Reque ... "}, Sabre\HTTP\Response {})
 6. /var/www/nextcloud/3rdparty/sabre/event/lib/EventEmitterTrait.php line 105
    call_user_func_array([Sabre\DAV\CorePlugin {},"httpPut"], [Sabre\HTTP\Requ ... }])
 7. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 479
    emit("method:PUT", [Sabre\HTTP\Requ ... }])
 8. /var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php line 254
    invokeMethod(Sabre\HTTP\Reque ... "}, Sabre\HTTP\Response {})
 9. /var/www/nextcloud/apps/dav/lib/Server.php line 301
    exec()
10. /var/www/nextcloud/apps/dav/appinfo/v2/remote.php line 35
    exec()
11. /var/www/nextcloud/remote.php line 163
    require_once("/var/www/nextcl ... p")

PUT /remote.php/dav/addressbooks/users/MYUSERNAME/contacts/FILENAME.vcf
from 192.168.0.14 by MYUSERNAME at 2019-10-24T17:26:03+00:00```

**Comment by AndyScherzinger** [comment]
@georgehrke @GretaD can this be fixed on the server side? If not which vCard version would be needed.

Also pinging @tobiasKaminsky for feedback as there miht be a good reason for the client to export to vCard 2.1.

**Comment by sephrat** [comment]
I have the same issue.
No contact is created on the server side after the sync. Manually importing the exported .vcf in the web interface generates said error.
No feedback is given in the app about an error in the process.

Android v3.8.1
NC server 17.0.0.

**Comment by Bugsbane** [comment]
@sephrat Do you get the same error message in your Nextcloud logs?

**Comment by sephrat** [comment]
> @sephrat Do you get the same error message in your Nextcloud logs?

It's actually in Nextcloud logs that I saw the error message (same as OP). There is no error message in the app.

**Comment by AndyScherzinger** [comment]
@georgehrke @GretaD @skjnldsv can this be fixed on the server side? If not which vCard version would be needed.

Also pinging @tobiasKaminsky for feedback as there miht be a good reason for the client to export to vCard 2.1.

**Comment by BernieO** [comment]
Nextcloud contacts does not support vCard2.1.
Minimum version is vCard 3.0: https://github.com/nextcloud/contacts/issues/990#issuecomment-563121615

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky please see latest comments. Server-side only support vCard 3.x+ while the Android clients exports 2.1, so we need to bump the export to 3.x+.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky checked the code, we get 2.1 from Android itself so we would need to port something like https://github.com/jowave/vcard2to3 (written to do this in python from converting vcfs from Android for Nextcloud)

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky Ezvcard should be able to do this, test implementation: #5181

@Bugsbane @sephrat @BernieO can you please give the QA build (should arrive in #5181 by tomorrow) a test drive and report back if this works for you? Any test&feedback is highly appreciated ❤️

**Comment by BernieO** [comment]
> can you please give the QA build (should arrive in #5181 by tomorrow) a test drive and report back if this works for you? Any test&feedback is highly appreciated ❤️

Sorry, I'd love to do that, but I don't use Android at all. I just stumbled over this searching for a solution for similar problem (https://github.com/BernieO/calcardbackup/issues/25) and thought I'd provide some useful information.

**Comment by AndyScherzinger** [comment]
@sephrat @Bugsbane fix will ship with 3.10 🎉
